src/DescriptorsCommand.cc

```diff
@@ -0,0 +1,902 @@
+        }
+
+        if (findDesc["status"] != 0) {
+            std::cerr << json_responses.toStyledString() << std::endl;
+            Json::Value return_error;
+            return_error["status"]  = RSCommand::Error;
+            return_error["info"]    = "Descriptors Not Found";
+            return error(return_error);
+        }
+    }
+    // Case (3)
+    else{
+
+        assert(json_responses.size() == 2);
+
+        // Get Set info.
+        const Json::Value& set_response = json_responses[0];
+
+        if (set_response["status"] != 0 ||
+            !set_response.isMember("entities")) {
+
+            Json::Value return_error;
+            return_error["status"]  = RSCommand::Error;
+            return_error["info"]    = "Set Not Found";
+            return error(return_error);
+        }
+
+        assert(set_response["entities"].size() == 1);
+
+        const Json::Value& set = set_response["entities"][0];
+
+        // This properties should always exist
+        assert(set.isMember(VDMS_DESC_SET_PATH_PROP));
+        assert(set.isMember(VDMS_DESC_SET_DIM_PROP));
+        std::string set_path = set[VDMS_DESC_SET_PATH_PROP].asString();
+        int dim = set[VDMS_DESC_SET_DIM_PROP].asInt();
+
+        if (!check_blob_size(blob, dim, 1)) {
+            Json::Value return_error;
+            return_error["status"] = RSCommand::Error;
+            return_error["info"] = "Blob (required) is null or size invalid";
+            return error(return_error);
+        }
+
+        std::vector<long>*  ids;
+        std::vector<float>* distances;
+
+        bool compute_distance = false;
+        if (results.isMember("list")) {
+
+            for (int i = 0; i < results["list"].size(); ++i) {
+                if (results["list"][i].asString() == "_distance") {
+                    compute_distance = true;
+                    break;
+                }
+
+            }
+        }
+
+        // Test whether there is any cached result.
+        assert (cache.isMember("cache_obj_id"));
+
+        long cache_obj_id = cache["cache_obj_id"].asInt64();
+
+        // Get from Cache
+        IDDistancePair& pair = _cache_map[cache_obj_id];
+        ids       = &pair.first;
+        distances = &pair.second;
+
+        findDesc = json_responses[1];
+
+        if (findDesc["status"] != 0 || !findDesc.isMember("entities")) {
+
+            Json::Value return_error;
+            return_error["status"]  = RSCommand::Error;
+            return_error["info"]    = "Descriptor Not Found in graph!";
+            return error(return_error);
+        }
+
+        Json::Value entities = findDesc["entities"];
+        findDesc.removeMember("entities");
+
+        convert_properties(entities);
+
+        std::cout << json_responses << std::endl;
+        std::cout << (*ids).size() << std::endl;
+
+        // This should never happen as ids will be have all (and only)
+        // the valid elements at this point
+        assert(entities.size() == (*ids).size());
```

Comment:
I know we were having an issue with this assert when all the features were identical within the vector. I know that they shouldn't all be identical, but is assert the right response? (Maybe we should throw here instead?)

Reply:
This assert does not control that, it controls that pmgd response is consistent with the information of the search. But since it is redundant because of previous controls, we can get rid of it.